File: backend/michicondrias_estilistas/app/models/grooming.py
import uuid
from sqlalchemy import Column, String, Text, Date, Time, Boolean, Float, DateTime
from sqlalchemy.sql import func
from app.db.session import Base

class GroomingFile(Base):
    __tablename__ = "grooming_files"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    pet_id = Column(String(36), nullable=False, index=True)
    hair_type = Column(String(100), nullable=True)
    preferred_shampoo = Column(String(100), nullable=True)
    behavior_notes = Column(Text, nullable=True)
    allergies_detected = Column(Text, nullable=True)
    last_service_date = Column(Date, nullable=True)


class GroomingAppointment(Base):
    __tablename__ = "grooming_appointments"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    groomer_id = Column(String(36), nullable=False, index=True)
    pet_id = Column(String(36), nullable=False, index=True)
    date = Column(Date, nullable=False)
    time = Column(Time, nullable=False)
    service_type = Column(String(100), nullable=False)
    status = Column(String(20), nullable=False, default="pending")
    before_photo_url = Column(String(500), nullable=True)
    after_photo_url = Column(String(500), nullable=True)
    skin_report = Column(Text, nullable=True)


class GroomingService(Base):
    __tablename__ = "grooming_services"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    groomer_id = Column(String(36), nullable=False, index=True)
    name = Column(String(150), nullable=False)
    description = Column(Text, nullable=True)
    price = Column(Float, nullable=False)
    duration_minutes = Column(Float, default=60.0)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: mobile/src/services/grooming.ts
import { apiFetch } from "../lib/api";

export interface GroomingAppointment {
    id: string;
    groomer_id: string;
    pet_id: string;
    date: string;
    time: string;
    service_type: string;
    status?: string;
    before_photo_url?: string;
    after_photo_url?: string;
    skin_report?: string;
}

export interface GroomingAppointmentCreate {
    groomer_id: string;
    pet_id: string;
    date: string;
    time: string;
    service_type: string;
    status?: string;
}

export interface GroomingAppointmentUpdatePhotos {
    before_photo_url?: string;
    after_photo_url?: string;
    status?: string;
    skin_report?: string;
}

export interface GroomingHistory {
    file?: GroomingFile;
    appointments: GroomingAppointment[];
}

export interface GroomingFile {
    id: string;
    pet_id: string;
    hair_type?: string;
    preferred_shampoo?: string;
    behavior_notes?: string;
    allergies_detected?: string;
    last_service_date?: string;
}

export interface GroomingService {
    id: string;
    groomer_id: string;
    name: string;
    description?: string;
    price: number;
    duration_minutes: number;
    is_active: boolean;
    created_at: string;
}

export interface GroomingServiceCreate {
    name: string;
    description?: string;
    price: number;
    duration_minutes?: number;
}

export async function createAppointment(data: GroomingAppointmentCreate): Promise<GroomingAppointment> {
    return apiFetch<GroomingAppointment>("estilistas", "/appointments", {
        method: "POST",
        body: JSON.stringify(data),
    });
}

export async function updateAppointmentPhotos(appointmentId: string, data: GroomingAppointmentUpdatePhotos): Promise<GroomingAppointment> {
    return apiFetch<GroomingAppointment>("estilistas", `/appointments/${appointmentId}/photos`, {
        method: "PUT",
        body: JSON.stringify(data),
    });
}

export async function getGroomingHistory(petId: string): Promise<GroomingHistory> {
    return apiFetch<GroomingHistory>("estilistas", `/files/${petId}`);
}

export async function createGroomingService(data: GroomingServiceCreate): Promise<GroomingService> {
    return apiFetch<GroomingService>("estilistas", "/services", {
        method: "POST",
        body: JSON.stringify(data),
    });
}

export async function getActiveServices(): Promise<GroomingService[]> {
    return apiFetch<GroomingService[]>("estilistas", "/services");
}

export async function getClientAppointments(): Promise<GroomingAppointment[]> {
    return apiFetch<GroomingAppointment[]>("estilistas", "/appointments/client");
}

export async function getProviderAppointments(): Promise<GroomingAppointment[]> {
    return apiFetch<GroomingAppointment[]>("estilistas", "/appointments/provider");
}

export async function getAvailableSlots(groomerId: string, targetDate: string): Promise<string[]> {
    const params = new URLSearchParams({ target_date: targetDate });
    return apiFetch<string[]>("estilistas", `/groomers/${groomerId}/available-slots?${params}`);
}
